cylc/flow/data_store_mgr.py

```diff
@@ -1025,30 +1023,30 @@ def insert_job(self, name, point_string, job_conf):
             stamp=f'{j_id}@{update_time}',
             id=j_id,
             submit_num=sub_num,
-            state=JOB_STATUSES_ALL[0],
+            state=status,
             task_proxy=tp_id,
-            job_runner_name=job_conf['job_runner_name'],
-            env_script=job_conf['env-script'],
-            err_script=job_conf['err-script'],
-            exit_script=job_conf['exit-script'],
-            execution_time_limit=job_conf['execution_time_limit'],
-            platform=job_conf['platform']['name'],
-            init_script=job_conf['init-script'],
-            post_script=job_conf['post-script'],
-            pre_script=job_conf['pre-script'],
-            script=job_conf['script'],
-            work_sub_dir=job_conf['work_d'],
             name=tproxy.name,
             cycle_point=tproxy.cycle_point,
-            directives=json.dumps(job_conf['directives']),
-            environment=json.dumps(job_conf['environment']),
-            param_var=json.dumps(job_conf['param_var'])
+            job_runner_name=job_conf.get('job_runner_name'),
```

Comment:
Allow these fields to be `None` for "prep-failed"-like jobs which did not get far enough through the process for these fields to be populated.